src/aggregation/metric/stats.rs

```diff
@@ -156,23 +156,36 @@ pub(crate) struct SegmentStatsCollector {
     field_type: Type,
     pub(crate) collecting_for: SegmentStatsType,
     pub(crate) stats: IntermediateStats,
+    pub(crate) accessor_idx: usize,
 }
 
 impl SegmentStatsCollector {
-    pub fn from_req(field_type: Type, collecting_for: SegmentStatsType) -> Self {
+    pub fn from_req(
+        field_type: Type,
+        collecting_for: SegmentStatsType,
+        accessor_idx: usize,
+    ) -> Self {
         Self {
             field_type,
             collecting_for,
             stats: IntermediateStats::default(),
+            accessor_idx,
         }
     }
-    pub(crate) fn collect_block(&mut self, docs: &[DocId], field: &Column<u64>) {
-        // TODO special case for Required, Optional column type
-        for doc in docs {
-            for val in field.values(*doc) {
+    pub(crate) fn collect_block_with_field(&mut self, docs: &[DocId], field: &Column<u64>) {
+        if field.get_cardinality() == Cardinality::Full {
+            for doc in docs {
+                let val = field.values.get_val(*doc);
                 let val1 = f64_from_fastfield_u64(val, &self.field_type);
```

Comment:
We should have N monomorphized loops for the different types. 
Even if it is const, the compiler does not know it can do it for us.

Reply:
Have you investigate if we would gain some perf if we brutally integrated the metrics computation into the ColumnValue?